ui/src/contexts/Application.jsx

```diff
@@ -86,10 +84,32 @@ function watchVault(id, dispatch, offerStatus) {
     window.localStorage.setItem(id, VaultStatus.CLOSED);
   }
 
-  vaultUpdater().catch(err => {
-    console.error('Vault watcher exception', id, err);
-    dispatch(updateVault({ id, vault: { status: VaultStatus.ERROR, err } }));
-  });
+  async function assetUpdater(asset) {
+    for await (const value of iterateNotifier(asset)) {
+      console.log('======== ASSET', id, value);
+      dispatch(
+        updateVault({
+          id,
+          vault: { asset: { ...value } },
```

Comment:
No need for destructuring:
```suggestion
          vault: { asset: value },
```
might want to get more specific on the name:
```suggestion
          vault: { asset: assetState },
```

Reply:
Done